### DISCUSSION TITLE
Anzeige der letzten Werte in Home Assistant

### DISCUSSION THREAD

**Discussion by copemande** [tbnobody/OpenDTU#229] [Q&A] [answered]
Hallo zusammen,

ich habe die OpenDTU problemlos bauen können und auch die Einbindung in Home Assistant funktioniert so weit, so gut. 

Leider werden in HA bei Dunkelheit, wenn der WR nicht mehr funkt, keine Werte mehr angezeigt bzw. der Sensor ist nicht mehr erreichbar. 

<img width="1046" alt="Capture1" src="https://user-images.githubusercontent.com/22273465/196001916-4a995065-e226-4b92-8a5a-416ba9a13ea1.PNG">

Das OpenDTU Web Interface zeigt mir folgendes:

<img width="1226" alt="Capture2" src="https://user-images.githubusercontent.com/22273465/196001996-ab6b5bbd-fbbe-4e6b-9657-daa07810693d.PNG">

Mir ist nicht klar, warum HA hier "unavailable", das Web Interface jedoch die letzten Werte anzeigt. Sobald es hell wird bekomme ich in HA auch wieder Werte. Kann mir vielleicht jemand einen Tip geben wo ich suchen muss? Ist es eine OpenDTU, MQTT, MQTT Broker oder HA Einstellung die geändert werden muß? Zumindest YieldDay und YieldTotal hätte ich gerne rund um die Uhr angezeigt.

Ich bedanke mich im Voraus.

(C)opeman

**Comment by battifreund** [comment, marked as answer]
Das liegt wahrscheinlich daran, dass OpenDTU standardmäßig das Attribut expire_after setzt. Du kannst das Verhalten ändern, indem Du in den MQTT-Einstellungen "Enable Expiration" ausschaltest. Dann behalten die Status die letzten Werte, bis die WR wieder neue Daten senden.

**Reply by copemande** [reply]
Danke battifreund für die schnelle Hilfe. Das wars :-)

**Reply by MaxiKing007** [reply]
Bin ich blind oder wo ist die Einstellung? 🙈

**Reply by copemande** [reply]
Guckst Du hier unter MQTT Settings:
![MQTT Settings](https://user-images.githubusercontent.com/22273465/197859461-c77590c6-41ce-4404-8252-4c26871c24d2.JPG)

**Reply by MaxiKing007** [reply]
Ahh ich nutze Homeassistant nicht und deshalb wird mir das nicht angezeigt :/
